src/Tasks/Resources/Strings.resx

```diff
@@ -221,6 +221,9 @@
 
         If this bucket overflows, pls. contact 'vsppbdev'.
   -->
+  <data name="Copy.LinklibraryFailedPrefix">
+    <value>The {0} library call failed with the following error code: </value>
```

Comment:
We usually try to support localizing order of arguments as well as the words in the message, so I think this should ideally be

```suggestion
    <value>The {0} library call failed with error code {1}</value>
```

Instead of having it be a prefix and appending after getting the localized string.

Here that looks like it might be complicated because of how the string is handled in `MakeSymbolicLink`. Could that take a format string as an input, instead of the current approach?

Reply:
> Could that take a format string as an input, instead of the current approach?

I think we can't do it like that. Since MakeSymbolicLink is defined in Framework and can't reference TaskLoggingHelper. So I pass the string errorMessagePrefix = ""(formated from the resource) to MakeSymbolicLink function. And real error code is  Marshal.GetLastWin32Error(), that returns the error code returned by the last unmanaged function symlink in MakeSymbolicLink.

Reply:
The alternative can be that `MakeSymbolicLink` would get back just the error code:

```
MakeSymbolicLink(string newFileName, string exitingFileName, ref string systemError)
```

Where `systemError` is either the descriptive HR string (on Win) or error code number (non-win) and the caller would then take care about wrapping it to logger-friendly, localized message (as caller has access to `TaskLoggingHelper`)